### PR TITLE
feat(@vtmn/react): add `VtmnRating` component

### PR DESCRIPTION
**Pull request by 0x2A-git**
## Changes description

This P.R aims to add the VtmnRating component

## Context

#366 

## Checklist

- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch**. Please, don't request directly from your main!
- [x] Check commits & PR names matches our requested structure. It must follow the https://www.conventionalcommits.org pattern.
- [x] Check your code additions will fail neither code linting checks.
- [x] I have reviewed the submitted code.
- [x] I have tested on related showcases.
- [x] If it includes design changes, please ask for a review with a core team designer.

## Does this introduce a breaking change?

- No

## Other information

The emphasis does not seem to be taken into account when rating is in edit mode.

### CODE REVIEW

**Review by lauthieb — CHANGES_REQUESTED** [review]

File: packages/sources/react/src/components/indicators/VtmnRating/VtmnRating.tsx [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+        >
+          {Array.from(Array(5).keys()).map((index) => (
+            <React.Fragment key={`rating-${index + 1}`}>
+              <input
+                onChange={(event: React.FormEvent<HTMLInputElement>) =>
+                  setValue(parseInt(event.currentTarget.value))
+                }
+                type="radio"
+                name={name}
+                id={`rating-${index + 1}`}
```
[the remark this answers is not in the document]
Comment (0x2A-git) [reply packages/sources/react/src/components/indicators/VtmnRating/VtmnRating.tsx]: You're totally right, I completely forgot to replace it with a dynamic id based on the name prop, it should now be fixed with 7c161ab :)

**Comment by Tlahey** [comment]
Hello @0x2A-git , we gonna make some changes on the `VtmnRating` after discuss with @lauthieb .
You can see the changes on this PR https://github.com/Decathlon/vitamin-web/pull/1121/files 😄

File: packages/showcases/react/stories/components/indicators/VtmnRating/VtmnRating.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  argTypes: {
+    ...argTypes,
+    comment: {
+      type: { name: 'number', required: false },
+      description:
+        'Comment displayed on the right side of the rating when using compact mode.',
+      defaultValue: false,
+      control: { type: 'number' },
+    },
+
```
Comment (lauthieb) [inline packages/showcases/react/stories/components/indicators/VtmnRating/VtmnRating.stories.tsx:21]: FYI, we just rework the `VtmnRating` component, `comments` is now available inside the component as a string property. See this PR: https://github.com/Decathlon/vitamin-web/pull/1121

I propose you globally check if all it's aligned as we did in the `VtmnRating` Svelte component here: https://github.com/Decathlon/vitamin-web/blob/main/packages/sources/svelte/src/components/indicators/VtmnRating/VtmnRating.svelte

**Review by lauthieb — CHANGES_REQUESTED** [review]

File: packages/showcases/react/stories/components/indicators/VtmnRating/VtmnRating.stories.tsx (hunk 2 above)
Comment (0x2A-git) [reply packages/showcases/react/stories/components/indicators/VtmnRating/VtmnRating.stories.tsx]: bf819f4 should now make the component behave the same way as the svelte one :)

**Review by lauthieb — APPROVED** [review]
Awesome Lucas @0x2A-git, thanks!!

**Pull request merged**